File: src/main/java/io/sundog/copy/manager/GovernCopy.java
package io.sundog.copy.manager;

import java.awt.*;
import java.awt.datatransfer.Clipboard;
import java.awt.datatransfer.StringSelection;

/**
 * コピーを司るクラス
 * @author SUNDOG
 *
 */
public class GovernCopy {

	/**
	 * 引数として受け取った文字列をクリップボードにコピーする
	 *
	 * @param text
	 *            コピーしたいテキスト
	 */
	public void doCopy(String text) {

		Clipboard cb = Toolkit.getDefaultToolkit().getSystemClipboard();
		StringSelection selection = new StringSelection(text);
		cb.setContents(selection, selection);

	}
}

File: composeApp/src/desktopMain/kotlin/org/sundog/copy/utils/GovernCopy.kt
package org.sundog.copy.utils

import java.awt.Toolkit
import java.awt.datatransfer.StringSelection

/**
 * コピーを司るクラス
 */
object GovernCopy {
    /**
     * 引数として受け取った文字列をクリップボードにコピーする
     */
    fun doCopy(text: String) {
        StringSelection(text).also {
            Toolkit.getDefaultToolkit().systemClipboard.setContents(it, it)
        }
    }
}